sunpy/coordinates/utils.py

```diff
@@ -265,3 +265,74 @@ def coordinates(self, points=None):
                         obstime=self.obstime,
                         observer=self.observer,
                         frame=Heliocentric).transform_to(self.start_frame)
+
+
+def get_rectangle_coordinates(bottom_left, *, top_right = None, width: u.deg = None, height: u.deg = None):
+    """
+    Specify a rectangular region of interest in spherical world coordinates.
+
+    Parameters
+    ----------
+    bottom_left : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The bottom_left coordinate of the rectangle. If a `SkyCoord` it can
+        have shape ``(2,)`` and also define ``top_right``.
+    top_right : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`
+        The top_right coordinate of the rectangle. Can only be omitted if
+        ``bottom_left`` has shape ``(2,)``.
+        If in a different frame than ``bottom_left`` and all required metadata for frame conversion is present,
+        ``top_right`` will be transformed to ``bottom_left`` frame.
+        Otherwise a ``ValueError`` will be incurred.
```

Comment:
```suggestion
        Otherwise a ``ValueError`` will be raised.
```
I don't think "incurred" is typical terminology.  This sentence could also be removed altogether.